src/MoBi.Presentation/Presenter/CreateSimulationPresenter.cs

```diff
@@ -117,13 +117,7 @@ private void validateDimensions(IModel model)
 
       private CreationResult createModel()
       {
-         //Create the model using a build configuration referencing the templates building block so that references to template builders are defined properly 
-         //we override the _buildConfiguration so that reference to builders are saved
-
-         // TODO SIMULATION_CONFIGURATION
-         // _simulationConfiguration = _buildConfigurationFactory.CreateFromReferencesUsedIn(_simulationConfiguration);
-         
-         _simulationConfiguration.SimulationSettings = _cloneManager.CloneBuildingBlock(_context.CurrentProject.SimulationSettings);
+         _simulationConfiguration.SimulationSettings = _cloneManager.CloneBuildingBlock(_context.CurrentProject.SimulationSettings ?? new SimulationSettings { Solver = new SolverSettings() });
```

Comment:
this is weird here.
Can this be null? If yes, why? 
If not, then not required.
Also don't we have a SimulationSettingsFactory that would set all the default for us? Presenter creating a bunch of domain objects that have nothing to do with it's purpose feels weird. In fact the whole retrieval of simulation settings belongs probably outside (if we want to separate concerns)

Reply:
It cannot be null. It was a patch I made to cover for a previous bug, which I then fixed, but then didn't remove the patch.